<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 39 Replication and Synthetic Data | A Guide on Data Analysis</title>
<meta name="author" content="Mike Nguyen">
<meta name="description" content="Access to comprehensive data is pivotal for replication, especially in the realm of social sciences. Yet, often the data are inaccessible, making replication a challenge (G. King 1995). This...">
<meta name="generator" content="bookdown 0.35 with bs4_book()">
<meta property="og:title" content="Chapter 39 Replication and Synthetic Data | A Guide on Data Analysis">
<meta property="og:type" content="book">
<meta property="og:url" content="https://bookdown.org/mike/data_analysis/replication-and-synthetic-data.html">
<meta property="og:image" content="https://bookdown.org/mike/data_analysis//images/cover.jpg">
<meta property="og:description" content="Access to comprehensive data is pivotal for replication, especially in the realm of social sciences. Yet, often the data are inaccessible, making replication a challenge (G. King 1995). This...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 39 Replication and Synthetic Data | A Guide on Data Analysis">
<meta name="twitter:description" content="Access to comprehensive data is pivotal for replication, especially in the realm of social sciences. Yet, often the data are inaccessible, making replication a challenge (G. King 1995). This...">
<meta name="twitter:image" content="https://bookdown.org/mike/data_analysis//images/cover.jpg">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.6.1/transition.js"></script><script src="libs/bs3compat-0.6.1/tabs.js"></script><script src="libs/bs3compat-0.6.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=GA_MEASUREMENT_ID"></script><script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){window.dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-DMNX2X65HQ');
    </script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">A Guide on Data Analysis</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="introduction.html"><span class="header-section-number">1</span> Introduction</a></li>
<li><a class="" href="prerequisites.html"><span class="header-section-number">2</span> Prerequisites</a></li>
<li class="book-part">I. BASIC</li>
<li><a class="" href="descriptive-stat.html"><span class="header-section-number">3</span> Descriptive Statistics</a></li>
<li><a class="" href="basic-statistical-inference.html"><span class="header-section-number">4</span> Basic Statistical Inference</a></li>
<li class="book-part">II. REGRESSION</li>
<li><a class="" href="linear-regression.html"><span class="header-section-number">5</span> Linear Regression</a></li>
<li><a class="" href="non-linear-regression.html"><span class="header-section-number">6</span> Non-linear Regression</a></li>
<li><a class="" href="generalized-linear-models.html"><span class="header-section-number">7</span> Generalized Linear Models</a></li>
<li><a class="" href="linear-mixed-models.html"><span class="header-section-number">8</span> Linear Mixed Models</a></li>
<li><a class="" href="nonlinear-and-generalized-linear-mixed-models.html"><span class="header-section-number">9</span> Nonlinear and Generalized Linear Mixed Models</a></li>
<li class="book-part">III. RAMIFICATIONS</li>
<li><a class="" href="model-specification.html"><span class="header-section-number">10</span> Model Specification</a></li>
<li><a class="" href="imputation-missing-data.html"><span class="header-section-number">11</span> Imputation (Missing Data)</a></li>
<li><a class="" href="data.html"><span class="header-section-number">12</span> Data</a></li>
<li><a class="" href="variable-transformation.html"><span class="header-section-number">13</span> Variable Transformation</a></li>
<li><a class="" href="hypothesis-testing.html"><span class="header-section-number">14</span> Hypothesis Testing</a></li>
<li><a class="" href="marginal-effects.html"><span class="header-section-number">15</span> Marginal Effects</a></li>
<li><a class="" href="prediction-and-estimation.html"><span class="header-section-number">16</span> Prediction and Estimation</a></li>
<li><a class="" href="moderation.html"><span class="header-section-number">17</span> Moderation</a></li>
<li class="book-part">IV. CAUSAL INFERENCE</li>
<li><a class="" href="causal-inference.html"><span class="header-section-number">18</span> Causal Inference</a></li>
<li class="book-part">A. EXPERIMENTAL DESIGN</li>
<li><a class="" href="experimental-design.html"><span class="header-section-number">19</span> Experimental Design</a></li>
<li><a class="" href="sampling.html"><span class="header-section-number">20</span> Sampling</a></li>
<li><a class="" href="analysis-of-variance-anova.html"><span class="header-section-number">21</span> Analysis of Variance (ANOVA)</a></li>
<li><a class="" href="multivariate-methods.html"><span class="header-section-number">22</span> Multivariate Methods</a></li>
<li class="book-part">B. QUASI-EXPERIMENTAL DESIGN</li>
<li><a class="" href="quasi-experimental.html"><span class="header-section-number">23</span> Quasi-experimental</a></li>
<li><a class="" href="regression-discontinuity.html"><span class="header-section-number">24</span> Regression Discontinuity</a></li>
<li><a class="" href="synthetic-difference-in-differences.html"><span class="header-section-number">25</span> Synthetic Difference-in-Differences</a></li>
<li><a class="" href="difference-in-differences.html"><span class="header-section-number">26</span> Difference-in-differences</a></li>
<li><a class="" href="synthetic-control.html"><span class="header-section-number">27</span> Synthetic Control</a></li>
<li><a class="" href="event-studies.html"><span class="header-section-number">28</span> Event Studies</a></li>
<li><a class="" href="matching-methods.html"><span class="header-section-number">29</span> Matching Methods</a></li>
<li><a class="" href="interrupted-time-series.html"><span class="header-section-number">30</span> Interrupted Time Series</a></li>
<li class="book-part">C. OTHER CONCERNS</li>
<li><a class="" href="endogeneity.html"><span class="header-section-number">31</span> Endogeneity</a></li>
<li><a class="" href="other-biases.html"><span class="header-section-number">32</span> Other Biases</a></li>
<li><a class="" href="controls.html"><span class="header-section-number">33</span> Controls</a></li>
<li class="book-part">V. MISCELLANEOUS</li>
<li><a class="" href="mediation.html"><span class="header-section-number">34</span> Mediation</a></li>
<li><a class="" href="directed-acyclic-graph.html"><span class="header-section-number">35</span> Directed Acyclic Graph</a></li>
<li><a class="" href="report.html"><span class="header-section-number">36</span> Report</a></li>
<li><a class="" href="exploratory-data-analysis.html"><span class="header-section-number">37</span> Exploratory Data Analysis</a></li>
<li><a class="" href="sensitivity-analysis-robustness-check.html"><span class="header-section-number">38</span> Sensitivity Analysis/ Robustness Check</a></li>
<li><a class="active" href="replication-and-synthetic-data.html"><span class="header-section-number">39</span> Replication and Synthetic Data</a></li>
<li class="book-part">APPENDIX</li>
<li><a class="" href="appendix.html"><span class="header-section-number">A</span> Appendix</a></li>
<li><a class="" href="bookdown-cheat-sheet.html"><span class="header-section-number">B</span> Bookdown cheat sheet</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/mikenguyen13/data_analysis">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="replication-and-synthetic-data" class="section level1" number="39">
<h1>
<span class="header-section-number">39</span> Replication and Synthetic Data<a class="anchor" aria-label="anchor" href="#replication-and-synthetic-data"><i class="fas fa-link"></i></a>
</h1>
<p>Access to comprehensive data is pivotal for replication, especially in the realm of social sciences. Yet, often the data are inaccessible, making replication a challenge <span class="citation">(<a href="references.html#ref-king1995replication">G. King 1995</a>)</span>. This chapter dives into the nuances of replication, the exceptions to its norms, and the significance of synthetic data.</p>
<div id="the-replication-standard" class="section level2" number="39.1">
<h2>
<span class="header-section-number">39.1</span> The Replication Standard<a class="anchor" aria-label="anchor" href="#the-replication-standard"><i class="fas fa-link"></i></a>
</h2>
<p>Replicability in research ensures:</p>
<ul>
<li>Credibility and comprehension of empirical studies.</li>
<li>Continuity and progression in the discipline.</li>
<li>Enhanced readership and academic citations.</li>
</ul>
<p>For a research to be replicable, the “replication standard” is vital: it entails providing all requisite information for replication by third parties. While quantitative research can, to some extent, offer clear data, qualitative studies pose complexities due to data depth.</p>
<div id="solutions-for-empirical-replication" class="section level3" number="39.1.1">
<h3>
<span class="header-section-number">39.1.1</span> Solutions for Empirical Replication<a class="anchor" aria-label="anchor" href="#solutions-for-empirical-replication"><i class="fas fa-link"></i></a>
</h3>
<ol style="list-style-type: decimal">
<li>
<strong>Role of Individual Authors</strong>:
<ul>
<li>Authors need to vouch for the replication standard for enhancing their work’s credibility.</li>
<li>Archives like the Inter-University Consortium for Political and Social Research (ICPSR) serve as depositories for replication datasets.</li>
</ul>
</li>
<li>
<strong>Creation of a Replication Data Set</strong>:
<ul>
<li>A public data set, consisting of both original and relevant complementary data, can serve replication purposes.</li>
</ul>
</li>
<li>
<strong>Professional Data Archives</strong>:
<ul>
<li>Organizations like ICPSR provide solutions to data storage and accessibility problems.</li>
</ul>
</li>
<li>
<strong>Educational Implications</strong>:
<ul>
<li>Replication can be an excellent educational tool, and many programs now emphasize its importance.</li>
</ul>
</li>
</ol>
</div>
<div id="free-data-repositories" class="section level3" number="39.1.2">
<h3>
<span class="header-section-number">39.1.2</span> Free Data Repositories<a class="anchor" aria-label="anchor" href="#free-data-repositories"><i class="fas fa-link"></i></a>
</h3>
<ol style="list-style-type: decimal">
<li><p><strong>Zenodo</strong>: Hosted by CERN, it provides a place for researchers to deposit datasets. It’s not subject-specific, so it caters to various disciplines.</p></li>
<li><p><strong>figshare</strong>: Allows researchers to upload, share, and cite their datasets.</p></li>
<li><p><strong>Dryad</strong>: Primarily for datasets associated with published articles in the biological and medical sciences.</p></li>
<li><p><strong>OpenICPSR</strong>: A public-facing version of the Inter-University Consortium for Political and Social Research (ICPSR) where researchers can deposit data without any cost.</p></li>
<li><p><strong>Harvard Dataverse</strong>: Hosted by Harvard University, this is an open-source repository software application dedicated to archiving, sharing, and citing research data.</p></li>
<li><p><strong>Mendeley Data</strong>: A multidisciplinary, free-to-use open access data repository where researchers can upload and share their datasets.</p></li>
<li><p><strong>Open Science Framework (OSF)</strong>: Offers both a platform for conducting research and a place to deposit datasets.</p></li>
<li><p><strong>PubMed Central</strong>: Specific to life sciences, but it’s an open repository for journal articles, preprints, and datasets.</p></li>
<li><p><strong>Registry of Research Data Repositories (re3data)</strong>: While not a repository itself, it provides a global registry of research data repositories from various academic disciplines.</p></li>
<li><p><strong>SocArXiv</strong>: An open archive for the social sciences.</p></li>
<li><p><strong>EarthArXiv</strong>: A preprints archive for earth science.</p></li>
<li><p><strong>Protein Data Bank (PDB)</strong>: For 3D structures of large biological molecules.</p></li>
<li><p><strong>Gene Expression Omnibus (GEO)</strong>: A public functional genomics data repository.</p></li>
<li><p><strong>The Language Archive (TLA)</strong>: Dedicated to data on languages worldwide, especially endangered languages.</p></li>
<li><p><strong>B2SHARE</strong>: A platform for storing and sharing research data sets in various disciplines, especially from European research projects.</p></li>
</ol>
</div>
<div id="exceptions-to-replication" class="section level3" number="39.1.3">
<h3>
<span class="header-section-number">39.1.3</span> Exceptions to Replication<a class="anchor" aria-label="anchor" href="#exceptions-to-replication"><i class="fas fa-link"></i></a>
</h3>
<p>Some exceptions to the replication standard are:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Confidentiality</strong>: Sometimes data, even fragmented, is too sensitive to share.</li>
<li>
<strong>Proprietary Data</strong>: Data sets owned by entities might restrict dissemination, but usually, parts of such data can still be shared.</li>
<li>
<strong>Rights of First Publication</strong>: Embargos might be set, but the essential data used in a study should be accessible.</li>
</ol>
</div>
</div>
<div id="synthetic-data-an-overview" class="section level2" number="39.2">
<h2>
<span class="header-section-number">39.2</span> Synthetic Data: An Overview<a class="anchor" aria-label="anchor" href="#synthetic-data-an-overview"><i class="fas fa-link"></i></a>
</h2>
<p>Synthetic data, modeling real data while ensuring anonymity, is becoming pivotal in research. While promising, it has its own complexities and should be approached with caution.</p>
<div id="benefits" class="section level3" number="39.2.1">
<h3>
<span class="header-section-number">39.2.1</span> Benefits<a class="anchor" aria-label="anchor" href="#benefits"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>Privacy preservation.</li>
<li>Data fairness and augmentation.</li>
<li>Acceleration in research.</li>
</ul>
</div>
<div id="concerns" class="section level3" number="39.2.2">
<h3>
<span class="header-section-number">39.2.2</span> Concerns<a class="anchor" aria-label="anchor" href="#concerns"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>Misconceptions about inherent privacy.</li>
<li>Challenges with data outliers.</li>
<li>Models relying solely on synthetic data can pose risks.</li>
</ul>
</div>
<div id="further-insights-on-synthetic-data" class="section level3" number="39.2.3">
<h3>
<span class="header-section-number">39.2.3</span> Further Insights on Synthetic Data<a class="anchor" aria-label="anchor" href="#further-insights-on-synthetic-data"><i class="fas fa-link"></i></a>
</h3>
<p>Synthetic data bridges the model-centric and data-centric perspectives, making it an essential tool in modern research. Analogously, it’s like viewing the Mona Lisa’s replica, with the real painting stored securely.</p>
<p>Future projects, such as utilizing the R’s diamonds dataset for synthetic data generation, hold promise in demonstrating the vast potentials of this technology.</p>
<p>For a deeper dive into synthetic data and its applications, refer to <span class="citation">(<a href="references.html#ref-jordon2022synthetic">Jordon et al. 2022</a>)</span>.</p>
</div>
</div>
<div id="application-15" class="section level2" number="39.3">
<h2>
<span class="header-section-number">39.3</span> Application<a class="anchor" aria-label="anchor" href="#application-15"><i class="fas fa-link"></i></a>
</h2>
<p>The easiest way to create synthetic data is to use the <code>synthpop</code> package. Alternatively, you can do it <a href="https://towardsdatascience.com/creating-synthetic-data-3774391c851d">manually</a></p>
<div class="sourceCode" id="cb732"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://www.synthpop.org.uk/">synthpop</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://easystats.github.io/performance/">performance</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># library(effectsize)</span></span>
<span><span class="co"># library(see)</span></span>
<span><span class="co"># library(patchwork)</span></span>
<span><span class="co"># library(knitr)</span></span>
<span><span class="co"># library(kableExtra)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Sepal.Length Sepal.Width Petal.Length Petal.Width Species</span></span>
<span><span class="co">#&gt; 1          5.1         3.5          1.4         0.2  setosa</span></span>
<span><span class="co">#&gt; 2          4.9         3.0          1.4         0.2  setosa</span></span>
<span><span class="co">#&gt; 3          4.7         3.2          1.3         0.2  setosa</span></span>
<span><span class="co">#&gt; 4          4.6         3.1          1.5         0.2  setosa</span></span>
<span><span class="co">#&gt; 5          5.0         3.6          1.4         0.2  setosa</span></span>
<span><span class="co">#&gt; 6          5.4         3.9          1.7         0.4  setosa</span></span>
<span></span>
<span><span class="fu">synthpop</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/synthpop/man/codebook.syn.html">codebook.syn</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span></span>
<span><span class="co">#&gt; $tab</span></span>
<span><span class="co">#&gt;       variable   class nmiss perctmiss ndistinct</span></span>
<span><span class="co">#&gt; 1 Sepal.Length numeric     0         0        35</span></span>
<span><span class="co">#&gt; 2  Sepal.Width numeric     0         0        23</span></span>
<span><span class="co">#&gt; 3 Petal.Length numeric     0         0        43</span></span>
<span><span class="co">#&gt; 4  Petal.Width numeric     0         0        22</span></span>
<span><span class="co">#&gt; 5      Species  factor     0         0         3</span></span>
<span><span class="co">#&gt;                             details</span></span>
<span><span class="co">#&gt; 1                  Range: 4.3 - 7.9</span></span>
<span><span class="co">#&gt; 2                    Range: 2 - 4.4</span></span>
<span><span class="co">#&gt; 3                    Range: 1 - 6.9</span></span>
<span><span class="co">#&gt; 4                  Range: 0.1 - 2.5</span></span>
<span><span class="co">#&gt; 5 'setosa' 'versicolor' 'virginica'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $labs</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span></span>
<span><span class="va">syn_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/synthpop/man/syn.html">syn</a></span><span class="op">(</span><span class="va">iris</span>, seed <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Synthesis</span></span>
<span><span class="co">#&gt; -----------</span></span>
<span><span class="co">#&gt;  Sepal.Length Sepal.Width Petal.Length Petal.Width Species</span></span>
<span></span>
<span><span class="co"># check for replciated uniques</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/synthpop/man/replicated.uniques.html">replicated.uniques</a></span><span class="op">(</span><span class="va">syn_df</span>, <span class="va">iris</span><span class="op">)</span></span>
<span><span class="co">#&gt; $replications</span></span>
<span><span class="co">#&gt;   [1]  TRUE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE  TRUE FALSE FALSE</span></span>
<span><span class="co">#&gt;  [13]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE</span></span>
<span><span class="co">#&gt;  [25] FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt;  [37] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt;  [49] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt;  [61] FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE  TRUE</span></span>
<span><span class="co">#&gt;  [73] FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE</span></span>
<span><span class="co">#&gt;  [85] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE  TRUE</span></span>
<span><span class="co">#&gt;  [97] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE FALSE</span></span>
<span><span class="co">#&gt; [109] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt; [121] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt; [133] FALSE  TRUE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt; [145] FALSE FALSE FALSE FALSE FALSE FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $no.uniques</span></span>
<span><span class="co">#&gt; [1] 148</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $no.replications</span></span>
<span><span class="co">#&gt; [1] 17</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $per.replications</span></span>
<span><span class="co">#&gt; [1] 11.33333</span></span>
<span></span>
<span></span>
<span><span class="co"># remove replicated uniques and adds a FAKE_DATA label </span></span>
<span><span class="co"># (in case a person can see his or own data in </span></span>
<span><span class="co"># the replicated data by chance)</span></span>
<span></span>
<span><span class="va">syn_df_sdc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/synthpop/man/sdc.html">sdc</a></span><span class="op">(</span><span class="va">syn_df</span>, <span class="va">iris</span>, </span>
<span>                  label <span class="op">=</span> <span class="st">"FAKE_DATA"</span>,</span>
<span>                  rm.replicated.uniques <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="co">#&gt; no. of replicated uniques: 17</span></span></code></pre></div>
<div class="sourceCode" id="cb733"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iris</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu">GGally</span><span class="fu">::</span><span class="fu"><a href="https://ggobi.github.io/ggally/reference/ggpairs.html">ggpairs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="39-rep_synthetic_data_files/figure-html/unnamed-chunk-2-1.png" width="90%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb734"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">syn_df</span><span class="op">$</span><span class="va">syn</span> <span class="op">|&gt;</span> </span>
<span>    <span class="fu">GGally</span><span class="fu">::</span><span class="fu"><a href="https://ggobi.github.io/ggally/reference/ggpairs.html">ggpairs</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="inline-figure"><img src="39-rep_synthetic_data_files/figure-html/unnamed-chunk-2-2.png" width="90%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb735"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lm_ori</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Sepal.Width</span> <span class="op">+</span> <span class="va">Petal.Length</span> , data <span class="op">=</span> <span class="va">iris</span><span class="op">)</span></span>
<span><span class="co"># performance::check_model(lm_ori)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">lm_ori</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; lm(formula = Sepal.Length ~ Sepal.Width + Petal.Length, data = iris)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residuals:</span></span>
<span><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span><span class="co">#&gt; -0.96159 -0.23489  0.00077  0.21453  0.78557 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt;              Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span><span class="co">#&gt; (Intercept)   2.24914    0.24797    9.07 7.04e-16 ***</span></span>
<span><span class="co">#&gt; Sepal.Width   0.59552    0.06933    8.59 1.16e-14 ***</span></span>
<span><span class="co">#&gt; Petal.Length  0.47192    0.01712   27.57  &lt; 2e-16 ***</span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residual standard error: 0.3333 on 147 degrees of freedom</span></span>
<span><span class="co">#&gt; Multiple R-squared:  0.8402, Adjusted R-squared:  0.838 </span></span>
<span><span class="co">#&gt; F-statistic: 386.4 on 2 and 147 DF,  p-value: &lt; 2.2e-16</span></span>
<span></span>
<span><span class="va">lm_syn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Sepal.Width</span> <span class="op">+</span> <span class="va">Petal.Length</span> , data <span class="op">=</span> <span class="va">syn_df</span><span class="op">$</span><span class="va">syn</span><span class="op">)</span></span>
<span><span class="co"># performance::check_model(lm_syn)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">lm_syn</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call:</span></span>
<span><span class="co">#&gt; lm(formula = Sepal.Length ~ Sepal.Width + Petal.Length, data = syn_df$syn)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residuals:</span></span>
<span><span class="co">#&gt;      Min       1Q   Median       3Q      Max </span></span>
<span><span class="co">#&gt; -0.79165 -0.22790 -0.01448  0.15893  1.13360 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt;              Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span><span class="co">#&gt; (Intercept)   2.96449    0.24538  12.081  &lt; 2e-16 ***</span></span>
<span><span class="co">#&gt; Sepal.Width   0.39214    0.06816   5.754  4.9e-08 ***</span></span>
<span><span class="co">#&gt; Petal.Length  0.45267    0.01743  25.974  &lt; 2e-16 ***</span></span>
<span><span class="co">#&gt; ---</span></span>
<span><span class="co">#&gt; Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Residual standard error: 0.3658 on 147 degrees of freedom</span></span>
<span><span class="co">#&gt; Multiple R-squared:  0.8246, Adjusted R-squared:  0.8222 </span></span>
<span><span class="co">#&gt; F-statistic: 345.6 on 2 and 147 DF,  p-value: &lt; 2.2e-16</span></span></code></pre></div>
<p>Open data can be assessed for its utility in two distinct ways:</p>
<ol style="list-style-type: decimal">
<li><p><strong>General Utility</strong>: This refers to the broad resemblances within the dataset, allowing for preliminary data exploration.</p></li>
<li><p><strong>Specific Utility</strong>: This focuses on the comparability of models derived from synthetic and original datasets, emphasizing analytical reproducibility.</p></li>
</ol>
<p>For General utility</p>
<div class="sourceCode" id="cb736"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/synthpop/man/compare.html">compare</a></span><span class="op">(</span><span class="va">syn_df</span>, <span class="va">iris</span><span class="op">)</span></span></code></pre></div>
<p>Specific utility</p>
<div class="sourceCode" id="cb737"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># just like regular lm, but for synthetic data</span></span>
<span><span class="va">lm_syn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/synthpop/man/glm.synds.html">lm.synds</a></span><span class="op">(</span><span class="va">Sepal.Length</span> <span class="op">~</span> <span class="va">Sepal.Width</span> <span class="op">+</span> <span class="va">Petal.Length</span> , data <span class="op">=</span> <span class="va">syn_df</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/synthpop/man/compare.html">compare</a></span><span class="op">(</span><span class="va">lm_syn</span>, <span class="va">iris</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call used to fit models to the data:</span></span>
<span><span class="co">#&gt; lm.synds(formula = Sepal.Length ~ Sepal.Width + Petal.Length, </span></span>
<span><span class="co">#&gt;     data = syn_df)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Differences between results based on synthetic and observed data:</span></span>
<span><span class="co">#&gt;              Synthetic  Observed        Diff Std. coef diff CI overlap</span></span>
<span><span class="co">#&gt; (Intercept)  2.9644900 2.2491402  0.71534988       2.884829  0.2640608</span></span>
<span><span class="co">#&gt; Sepal.Width  0.3921429 0.5955247 -0.20338187      -2.933611  0.2516161</span></span>
<span><span class="co">#&gt; Petal.Length 0.4526695 0.4719200 -0.01925058      -1.124602  0.7131064</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Measures for one synthesis and 3 coefficients</span></span>
<span><span class="co">#&gt; Mean confidence interval overlap:  0.4095944</span></span>
<span><span class="co">#&gt; Mean absolute std. coef diff:  2.314347</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Mahalanobis distance ratio for lack-of-fit (target 1.0): 3.08</span></span>
<span><span class="co">#&gt; Lack-of-fit test: 9.23442; p-value 0.0263 for test that synthesis model is</span></span>
<span><span class="co">#&gt; compatible with a chi-squared test with 3 degrees of freedom.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Confidence interval plot:</span></span></code></pre></div>
<div class="inline-figure"><img src="39-rep_synthetic_data_files/figure-html/unnamed-chunk-5-1.png" width="90%" style="display: block; margin: auto;"></div>
<div class="sourceCode" id="cb738"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># summary(lm_syn)</span></span></code></pre></div>
<p>You basically want your lack-of-fit test to be non-significant.</p>

</div>
</div>



<div class="chapter-nav">
<div class="prev"><a href="sensitivity-analysis-robustness-check.html"><span class="header-section-number">38</span> Sensitivity Analysis/ Robustness Check</a></div>
<div class="next"><a href="appendix.html"><span class="header-section-number">A</span> Appendix</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#replication-and-synthetic-data"><span class="header-section-number">39</span> Replication and Synthetic Data</a></li>
<li>
<a class="nav-link" href="#the-replication-standard"><span class="header-section-number">39.1</span> The Replication Standard</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#solutions-for-empirical-replication"><span class="header-section-number">39.1.1</span> Solutions for Empirical Replication</a></li>
<li><a class="nav-link" href="#free-data-repositories"><span class="header-section-number">39.1.2</span> Free Data Repositories</a></li>
<li><a class="nav-link" href="#exceptions-to-replication"><span class="header-section-number">39.1.3</span> Exceptions to Replication</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#synthetic-data-an-overview"><span class="header-section-number">39.2</span> Synthetic Data: An Overview</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#benefits"><span class="header-section-number">39.2.1</span> Benefits</a></li>
<li><a class="nav-link" href="#concerns"><span class="header-section-number">39.2.2</span> Concerns</a></li>
<li><a class="nav-link" href="#further-insights-on-synthetic-data"><span class="header-section-number">39.2.3</span> Further Insights on Synthetic Data</a></li>
</ul>
</li>
<li><a class="nav-link" href="#application-15"><span class="header-section-number">39.3</span> Application</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/mikenguyen13/data_analysis/blob/main/39-rep_synthetic_data.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/mikenguyen13/data_analysis/edit/main/39-rep_synthetic_data.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>
</div>
  

  

</div>
 <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>A Guide on Data Analysis</strong>" was written by Mike Nguyen. It was last built on 2024-06-06.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
